### PR TITLE
feat: welcome back screen [branch ch12939]

### PR DESCRIPTION
**Pull request by Karim94**
#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/12939/mobile-welcome-back-screen
#### Testing instructions  
Entering the app after signing out from an account should show the "Welcome Back" screen, with the username and firstname of the last account. You can press switch user to show the regular sign in screen without any saved info, or go back to 'Welcome" screen (the one with the illustration) by pressing the back arrow button. Either one of those actions clears the last user, so restarting the app again will take the user to the "Welcome screen" with the illustration.

Here are all the possible combinations of scenarios and their expected results:

1- Opening app "fresh" (for the first time, or without having previously logged in)
Expected: See Welcome screen (the one with the illustration)

2- Sign in -> refresh app
Expected: User is signed in

3- Sign in -> logout -> app restarts -> open app
Expected: See Welcome Back Screen (Welcome back, *Name*)

4- Repeat step 3 -> after app is opened and welcome back screen is visible -> refresh app
Expected: See Welcome Back Screen (Welcome back, *Name*)

5- Repeat step 3 -> after app is opened and welcome back screen is visible -> press back button
Expected: See Welcome Screen (illustration)

6- Repeat step 5 -> after seeing Welcome Screen -> refresh
Expected: Should see Welcome Screen again

7- Repeat step 3 -> after app is opened and welcome back screen is visible -> press "change user"
Expected: See Welcome Back Screen (Welcome back, NO name)

7- Repeat step 7 -> refresh app
Expected: See Welcome Screen (illustration)

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: app/components/App.js [hunk 1]
```diff
@@ -80,13 +80,16 @@ export default class App extends SafeComponent {
     async componentWillMount() {
         if (!MockComponent) {
             let route = routerApp.routes.loading;
-            if (!await User.getLastAuthenticated()
-                && !await TinyDb.system.getValue('apple-review-login')) {
+            const lastUser = await User.getLastAuthenticated();
+
+            if (lastUser) route = routerApp.routes.loginWelcomeBack;
```
Comment (seavan) [inline app/components/App.js]: you always get transitioned to welcome back screen. thus, you never get "loading" screen.

after signing up and relaoding the app, insetad of getting loading screen and auto-login, I get the welcome back screen.

this is an incorrect behaviour.

**Comment by ebilg** [comment]
can you add a progress indicator when signing in from the welcome back screen? currently app looks frozen

**Comment by seavan** [comment]
@ebilg added.

**Comment by ebilg** [comment]
can be merged

**Comment by seavan** [comment]
@ebilg there was a bit of rectoring done before merge. please take a look at the whole signup/login on RC, too.

**Pull request merged**